### ISSUE TITLE
[BUG] Multi data source test url is a fake endpoint

### ISSUE THREAD

**Issue by BionIT** [opensearch-project/opensearch-dashboards-functional-test#1202]
### What is the bug?
The OSD_TEST_DOMAIN_ENDPOINT_URL used in cypress tests, eg. https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/integration/core-opensearch-dashboards/opensearch-dashboards/datasource-management-plugin/1_create_datasource.spec.js#L8C3-L8C31 is `https://opensearch.org` which is not actual data source endpoint (see https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/dashboards/datasource-management-dashboards-plugin/constants.js#L7). Test connection would fail

### How can one reproduce the bug?
Check the link above 

### What is the expected behavior?
We should use the port from docker container or a read endpoint with credentials passed via secrets

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

**Comment by zhongnansu** [comment]
Since we can't use an real permanent endpoint, we probably need to find a way to spin up an OS cluster through cypress, and use that as datasource.

**Comment by seraphjiang** [comment]
> Since we can't use an real permanent endpoint, we probably need to find a way to spin up an OS cluster through cypress, and use that as datasource.

Why not create one and keep reusing it

**Comment by zhongnansu** [comment]
> > Since we can't use an real permanent endpoint, we probably need to find a way to spin up an OS cluster through cypress, and use that as datasource.
> 
> Why not create one and keep reusing it

@seraphjiang Do you mean for example we create an Amazon OpenSearch domain and reuse this real endpoint as datasource in cypress test? Since cypress tests might need to read/write data in the datasource, we can't let everyone have open access to a public domain

**Comment by BionIT** [comment]
@zhongnansu Since Derek's change has been merged in, can we use the workflow he added to spin off opensearch cluster and use the local cluster as the data source? I saw security plugin already uses it in their cypress test

**Comment by zhongnansu** [comment]
> @zhongnansu Since Derek's change has been merged in, can we use the workflow he added to spin off opensearch cluster and use the local cluster as the data source? I saw security plugin already uses it in their cypress test

@BionIT Yes, we can. However for current test suites of multiple datasource, we don't have any test that actually requires an endpoint to be valid. I can work on adding some more test maybe around test data source connection, which needs a valid endpoint.

**Comment by kavilla** [comment]
@BionIT is this for 2.14 or completed?

**Comment by kavilla** [comment]
Pinged @BionIT, stated good to move to 2.15.
